### PR TITLE
Allow customizing error reports before sending to Bugsnag

### PR DESCRIPTION
**Pull request by aaronrenner**
This was heavily inspired by @mootpointer's work on #12 to add context when reporting to bugsnag, but also allows the developer to customize the context that's sent.

I added an example to the readme as well as a couple examples in the tests. Please let me know what you think. I'd be happy to make any adjustments to ensure this is a good, extensible api for developers.

### CODE REVIEW

**Comment by mootpointer** [comment]
This looks handy. One thing that I was looking at while building my implementation was the ability to not send some params. For example, we probably don't want to be sending sensitive authentication data to Bugsnag from params.

**Comment by aaronrenner** [comment]
Good point. Do you think it would be better to not send params by default, or have a way to filter sensitive parameters?

**Comment by mootpointer** [comment]
I think before we can send params (which tend to be pretty important in debugging) we need a way to filter them.

**Review by mootpointer — CHANGES_REQUESTED** [review]
I've done some work here https://github.com/mootpointer/plugsnag/commit/156d24a40cc29dfbeb5343570443e7e91ffead3e which allows the filtering of parameters before sending to Bugsnag. It would be good to merge that in (if you want, you can [give me commit access to this PR](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) and I'll just add the commit) before we proceed with this PR.

Apart from I quite like the architecture that you've brought in here.

**Comment by aaronrenner** [comment]
Thanks for continuing to work on this. I've gotten busy and pulled off on other work. Your changes look great and it would probably be good to be able to filter request headers as well for things like api keys.

I just gave you edit access to the PR so you can merge in your changes. Thanks for all you do to maintain this open source library. I know they are a lot of work! 😄

File: lib/plugsnag.ex
Comment (aaronrenner) [inline lib/plugsnag.ex:31]: Probably would be more intention revealing if I used `Map.from_struct/1`

**Review by mootpointer — APPROVED** [review]

**Pull request merged**
